exam/models/exam.py

```diff
@@ -31,30 +31,59 @@ class ExtendedTimeTable(models.Model):
     _inherit = 'time.table'
 
     @api.multi
-    @api.depends('timetable_ids')
-    def _compute_users(self):
-        '''Method to compute user'''
+    def unlink(self):
+        exam = self.env['exam.exam']
+        schedule_line = self.env['exam.schedule.line']
         for rec in self:
-            rec.user_ids = [teacher.teacher_id.user_id.id
-                            for teacher in rec.timetable_ids
-                            if rec.timetable_type == 'exam']
-        return True
+            exam_search = exam.search([('state', '=', 'running')])
+            for data in exam_search:
+                schedule_line_search = schedule_line.search([('exam_id', '=',
+                                                              data.id),
+                                                             ('timetable_id',
+                                                              '=', rec.id)])
+                if schedule_line_search:
+                    raise ValidationError(_('''You cannot delete schedule
+                                           of exam in running state'''))
+        return super(ExtendedTimeTable, self).unlink()
 
     timetable_type = fields.Selection(selection_add=[('exam', 'Exam')],
                                       string='Time Table Type', required=True,
                                       inivisible=False)
     exam_timetable_line_ids = fields.One2many('time.table.line', 'table_id',
                                               'TimeTable')
     exam_id = fields.Many2one('exam.exam', 'Exam')
-    users = fields.Many2many('res.users', string="Users",
-                             compute="_compute_users", store=True)
```

Comment:
Explain the purpose of removing the field.

Reply:
We have inherited model time.table in exam module and user_ids field is defined in the model time.table.So it is removed from exam.py file